FAILURE: build_docker_java

Saw this on #18746 

https://source.cloud.google.com/results/invocations/33128db9-ffe8-4dc8-bc23-142bdd05bc22/targets

```
> Task :grpc-interop-testing:jar
> Task :grpc-interop-testing:reconnect_test_client
> Task :grpc-interop-testing:startScripts SKIPPED
> Task :grpc-interop-testing:stresstest_client
> Task :grpc-interop-testing:test_client
> Task :grpc-interop-testing:test_server
> Task :grpc-interop-testing:installDist
BUILD SUCCESSFUL in 3m 16s
51 actionable tasks: 51 executed
/
rm: cannot remove '/root/.gradle': Directory not empty
2019-04-26 13:52:05,851 FAILED: build_docker_java [ret=1, pid=10133, time=347.2sec]
```

seeing this fail occasionally with e.g. "connection reset" errors while downloading dependencies 

see e.g.:

```
++ uuidgen
+ CONTAINER_NAME=build_grpc_interop_java_1fa1dc6b-7aa4-46e6-b029-03957fe1dd92
+ docker run --cap-add SYS_PTRACE -e 'THIS_IS_REALLY_NEEDED=see https://github.com/docker/docker/issues/14203 for why docker is awful' -e 'THIS_IS_REALLY_NEEDED_ONCE_AGAIN=For issue 4835. See https://github.com/docker/docker/issues/14203 for why docker is awful' -i -v /tmpfs/src/github/grpc:/var/local/jenkins/grpc:ro -v /tmpfs/src/github/grpc-java:/var/local/jenkins/grpc-java:ro -v /tmpfs/src/github/grpc-go:/var/local/jenkins/grpc-go:ro -v /tmpfs/src/github/grpc-dart:/var/local/jenkins/grpc-dart:ro -v /tmpfs/src/github/grpc-node:/var/local/jenkins/grpc-node:ro -v /tmpfs/src/github/grpc-dotnet:/var/local/jenkins/grpc-dotnet:ro -v /home/kbuilder/service_account:/var/local/jenkins/service_account:ro --name=build_grpc_interop_java_1fa1dc6b-7aa4-46e6-b029-03957fe1dd92 grpctesting/grpc_interop_java_c0dc2dddbb3c7b62b4b8e059571c5181af5bcc0f bash -l /var/local/jenkins/grpc/tools/dockerfile/interoptest/grpc_interop_java/build_interop.sh
stdin: is not a tty
/tmp/grpc-java /
Downloading https://services.gradle.org/distributions/gradle-5.6.2-bin.zip
.........................................................................................

Welcome to Gradle 5.6.2!

Here are the highlights of this release:
 - Incremental Groovy compilation
 - Groovy compile avoidance
 - Test fixtures for Java projects
 - Manage plugin versions via settings script

For more details see https://docs.gradle.org/5.6.2/release-notes.html

*** Skipping the build of codegen and compilation of proto files because skipCodegen=true
  * Skipping the build of Android projects because skipAndroid=true

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'grpc'.
> Could not resolve all artifacts for configuration ':classpath'.
   > Could not download databinding-compiler-common.jar (androidx.databinding:databinding-compiler-common:3.3.0)
      > Could not get resource 'https://dl.google.com/dl/android/maven2/androidx/databinding/databinding-compiler-common/3.3.0/databinding-compiler-common-3.3.0.jar'.
         > Connection reset

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 53s

2020-03-02 01:15:57,870 FAILED: build_docker_java [ret=1, pid=14558, time=239.2sec]
```

Seen again:

```
Welcome to Gradle 5.6.2!

Here are the highlights of this release:
 - Incremental Groovy compilation
 - Groovy compile avoidance
 - Test fixtures for Java projects
 - Manage plugin versions via settings script

For more details see https://docs.gradle.org/5.6.2/release-notes.html

*** Skipping the build of codegen and compilation of proto files because skipCodegen=true
  * Skipping the build of Android projects because skipAndroid=true

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'grpc'.
> Could not resolve all artifacts for configuration ':classpath'.
   > Could not download ddmlib.jar (com.android.tools.ddms:ddmlib:26.5.0)
      > Could not get resource 'https://dl.google.com/dl/android/maven2/com/android/tools/ddms/ddmlib/26.5.0/ddmlib-26.5.0.jar'.
         > Connection reset

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 32s
```

https://source.cloud.google.com/results/invocations/47ec9690-c4cc-4f8d-88b3-c4883cbc000b/targets/grpc%2Fcore%2Fpull_request%2Flinux%2Fgrpc_interop_toprod/log

```

Welcome to Gradle 5.6.2!

Here are the highlights of this release:
 - Incremental Groovy compilation
 - Groovy compile avoidance
 - Test fixtures for Java projects
 - Manage plugin versions via settings script

For more details see https://docs.gradle.org/5.6.2/release-notes.html

*** Skipping the build of codegen and compilation of proto files because skipCodegen=true
  * Skipping the build of Android projects because skipAndroid=true

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'grpc'.
> Could not resolve all artifacts for configuration ':classpath'.
   > Could not download builder.jar (com.android.tools.build:builder:3.5.0)
      > Could not get resource 'https://dl.google.com/dl/android/maven2/com/android/tools/build/builder/3.5.0/builder-3.5.0.jar'.
         > Connection reset

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 36s
```

https://source.cloud.google.com/results/invocations/9d5f40b9-c05d-4321-9d15-6abee6ae0388/targets/grpc%2Fcore%2Fpull_request%2Flinux%2Fgrpc_interop_toprod/log

Happened again:

```
Downloading https://services.gradle.org/distributions/gradle-5.6.2-bin.zip
.........................................................................................

Welcome to Gradle 5.6.2!

Here are the highlights of this release:
 - Incremental Groovy compilation
 - Groovy compile avoidance
 - Test fixtures for Java projects
 - Manage plugin versions via settings script

For more details see https://docs.gradle.org/5.6.2/release-notes.html

*** Skipping the build of codegen and compilation of proto files because skipCodegen=true
  * Skipping the build of Android projects because skipAndroid=true

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'grpc'.
> Could not resolve all artifacts for configuration ':classpath'.
   > Could not download sdk-common.jar (com.android.tools:sdk-common:26.5.0)
      > Could not get resource 'https://dl.google.com/dl/android/maven2/com/android/tools/sdk-common/26.5.0/sdk-common-26.5.0.jar'.
         > Connection reset

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 41s
```

https://source.cloud.google.com/results/invocations/a0f46588-eadf-4cd1-ae77-e5c397d974f3/targets/github%2Fgrpc%2Finterop_docker_build/tests

cc @ejona86 @jtattermusch - just an idea is it possible to cache any of the Java interop binary dependencies into the pre-built interop docker images, rather than fetch them per-run?

--- a/templates/tools/dockerfile/java_build_interop.sh.include
+++ b/templates/tools/dockerfile/java_build_interop.sh.include
@@ -22,7 +22,9 @@ cp -r /var/local/jenkins/grpc-java /tmp/grpc-java
 cp -r /var/local/jenkins/service_account $HOME || true
 
 pushd /tmp/grpc-java
-./gradlew --no-daemon :grpc-interop-testing:installDist -PskipCodegen=true -PskipAndroid=true
+# make two attempts; downloads can fail. See https://github.com/grpc/grpc/issues/18892
+./gradlew --no-daemon :grpc-interop-testing:installDist -PskipCodegen=true -PskipAndroid=true || ${'\\'}
+    ./gradlew --no-daemon :grpc-interop-testing:installDist -PskipCodegen=true -PskipAndroid=true
 
 mkdir -p /var/local/git/grpc-java/
 cp -r --parents -t /var/local/git/grpc-java/ ${'\\'}

--- a/tools/dockerfile/interoptest/grpc_interop_java/build_interop.sh
+++ b/tools/dockerfile/interoptest/grpc_interop_java/build_interop.sh
@@ -22,7 +22,9 @@ cp -r /var/local/jenkins/grpc-java /tmp/grpc-java
 cp -r /var/local/jenkins/service_account $HOME || true
 
 pushd /tmp/grpc-java
-./gradlew --no-daemon :grpc-interop-testing:installDist -PskipCodegen=true -PskipAndroid=true
+# make two attempts; downloads can fail. See https://github.com/grpc/grpc/issues/18892
+./gradlew --no-daemon :grpc-interop-testing:installDist -PskipCodegen=true -PskipAndroid=true || \
+    ./gradlew --no-daemon :grpc-interop-testing:installDist -PskipCodegen=true -PskipAndroid=true
 
 mkdir -p /var/local/git/grpc-java/
 cp -r --parents -t /var/local/git/grpc-java/ \

